src/avram/save_operation.cr

```diff
@@ -248,17 +261,16 @@ abstract class Avram::SaveOperation(T)
     _changes
   end
 
-  private def cast_value(value : Nil)
-    nil
-  end
+  macro add_cast_value_methods(columns)
+    private def cast_value(value : Nil)
+      nil
+    end
 
-  private def cast_value(value : Object)
-    case value
-    when JSON::Any
-      value.to_json
-    else
-      value.to_s
+    {% for column in columns %}
+    private def cast_value(value : {{ column[:type] }})
+      value.not_nil!.class.adapter.to_db(value.as({{ column[:type] }}))
```

Comment:
Pass the `value` to it's `Lucky.to_db` for parsing. In most cases it's just calling `to_s` on the value, but in the case of the Arrays, it passes to `PQ::Param` to properly encode so `[true]` becomes `{t}`, and so on. 

We cast the value here with `value.as({{ column[:type] }})` otherwise the compiler sees a Union and throws no method overload. Also, I had to add the `not_nil!` because the compiler kept throwing undefined method adapter for Nil.class. In theory, it should never be nil because of the other `cast_value` overload, but maybe that would have to be defined later?

Reply:
This makes sense. I think it would be good to include this as a comment in the code. Usually I'm anti-comments but in this case I think it would be handy